### ISSUE TITLE
print_debug prints in release mode

### ISSUE THREAD

**Issue by MegaMasta** [godotengine/godot#48068]
**Godot version:**

v3.2.3.stable.official

**OS/device including version:**

Windows 10 64-bit Build 19041.928

**Issue description:**

Using [print_debug](https://docs.godotengine.org/en/stable/classes/class_@gdscript.html#class-gdscript-method-print-debug) in an exported build with mode "Release" prints messages.
According to the docs, it is expected that `print_debug` only prints when exported in debug mode or ran from the IDE.

**Steps to reproduce:**
* In GDScript use `print_debug("anything")`
* (If not done already:) Import export templates (download from https://www.godotengine.org/download)
* Export with mode "Release" (respectively "Export with Debug" disabled)
* Run the exported program and look at the output (for example, in a Windows build redirect stdout to a file).
-> It prints "anything"

**Minimal reproduction project:**

[Test_print_debug.zip](https://github.com/godotengine/godot/files/6354201/Test_print_debug.zip)
Note: In this project, there is also a comparison code using [OS.is_debug_build](https://docs.godotengine.org/en/stable/classes/class_os.html?highlight=is_debug_build#class-os-method-is-debug-build), which works correctly.

**Comment by Calinou** [comment]
It's possible that this only happens with file logging, rather than stdout. However, you can't inspect a program's stdout on Windows if it was compiled with the GUI subsystem (rather than the console subsystem). Release export templates are compiled with the GUI subsystem, unlike the editor and debug export templates which are compiled with the console subsystem.

**Comment by Sslaxx** [comment]
Can confirm this under Linux (Ubuntu 20.04) with 3.3 stable.

Using release mode, I get this output on the terminal:

```
OpenGL ES 2.0 Renderer: GeForce GTX 1650/PCIe/SSE2
OpenGL ES Batching: ON
 
res://Scripts/Singletons/helper_functions.gd ready. Node [Node:1163] (helper_functions).
res://Scripts/Singletons/random_helpers.gd ready. Node [Node:1168] (random_helpers).
RNG initialised; seed is 8698224463306719903.
res://Scripts/Singletons/sound_player.gd ready. Node [AudioStreamPlayer:1175] (sound_player).
res://Scripts/Singletons/music_player.gd ready. Node [AudioStreamPlayer:1180] (music_player).
res://Scripts/Singletons/jingle_player.gd ready. Node [AudioStreamPlayer:1185] (jingle_player).
res://Scripts/scn_preload.gd ready. Node [Node2D:1206] (Preloader).
Current sound_library contents: {no_sound:[AudioStreamOGGVorbis:1172], select:[AudioStreamOGGVorbis:1212]}
Current sound_library contents: {lose_influence:[AudioStreamOGGVorbis:1214], no_sound:[AudioStreamOGGVorbis:1172], select:[AudioStreamOGGVorbis:1212]}
Current sound_library contents: {gain_influence:[AudioStreamOGGVorbis:1216], lose_influence:[AudioStreamOGGVorbis:1214], no_sound:[AudioStreamOGGVorbis:1172], select:[AudioStreamOGGVorbis:1212]}
AAA: snd_sfx_ring.ogg
res://Scripts/main_menu.gd ready. Node [Control:1229] (main_menu).
Playing [AudioStreamOGGVorbis:1250] from res://Assets/Audio/Music/01-Title.ogg, offset 0.
```

**Comment by icanhaz** [comment]
> It's possible that this only happens with file logging, rather than stdout logging.

It occurs in HTML 5 release exports, too. The messages are printed in the console.

**Comment by MegaMasta** [comment]
According to the Pull Request #18966 and the implemented feature issues #3884 and #18319, it wasn't intended that print_debug only prints in debug mode, but rather prints the file and line of the code where the print happens. And it seems like that extra debug information is only printed in debug mode.

More importantly, there was an issue #6253, where this feature (only printing in debug mode) was rejected.
> Since it is possible to easily implement this feature by adding a very short function to a singleton, I'm closing this. (It is also more flexible this way.)

The doc to `print_debug` was added in #44911, 2.5 years later.

So for me the doc ist just wrong.
Should the doc be fixed?

**Comment by Sslaxx** [comment]
Note that `print_debug` did not exist at the time I opened #6253. #18966 makes no referral to when `print_debug` should work. I would assume the documentation to be correct in this instance.
